### ISSUE TITLE
TextPainter is not working well for WidgetSpan in web

### ISSUE THREAD

**Issue by zmtzawqlp** [flutter/flutter#47491]
TextPainter is not wroking well for WidgetSpan higher than v1.8.5(Not Exactly)

* lower than v1.8.5

- [x]  TextPainter .getPositionForOffset
- [ ]  TextPainter .getOffsetForCaret
- [x]  TextPainter .getBoxesForSelection

* higher than v1.8.5

- [ ]  TextPainter .getPositionForOffset
- [ ]  TextPainter .getOffsetForCaret
- [x]  TextPainter .getBoxesForSelection

This issue is refer to https://github.com/flutter/flutter/issues/38474. 

I'm look forward to fixing this issue asap, it's important to us who want to select WidgetSpan.

## Steps to Reproduce

click button to print info. please run with attached files ( paragraph.dart and rich_text.dart are same as official but public textPainter from RenderParagraph)
[lib.zip](https://github.com/flutter/flutter/files/3986418/lib.zip)

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';
import 'rich_text.dart' as rich_text;
import 'paragraph.dart' as paragraph;

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Find Second WidgetSpan'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  GlobalKey key = GlobalKey();
  String _printS = "";

  @override
  void initState() {
    super.initState();
  }

  void _print() {
    setState(() {
      final textPainter =
          (key.currentContext.findRenderObject() as paragraph.RenderParagraph)
              .textPainter;
      //try to find TextPosition of second WidgetSpan
      final TextPosition textPosition =
          textPainter.getPositionForOffset(Offset(150.0, 50.0));
      //try to find position of second WidgetSpan
      final Offset offset = textPainter.getOffsetForCaret(
          TextPosition(offset: 1, affinity: TextAffinity.upstream), Rect.zero);
      //try to find selection boxes of second WidgetSpan
      final List<TextBox> boxes = textPainter
          .getBoxesForSelection(TextSelection(baseOffset: 1, extentOffset: 2));

      _printS =
          "TextPosition：${textPosition.offset} \n\nOffset: $offset \n\nSelectionTextBox: ${boxes.first}";
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Column(
        children: <Widget>[
          Container(
            height: 200.0,
            child: Text(_printS),
          ),
          Expanded(
            child: rich_text.RichText(
              key: key,
              text: TextSpan(children: <InlineSpan>[
                WidgetSpan(
                    child: Container(
                  color: Colors.red,
                  width: 100.0,
                  height: 100.0,
                )),
                WidgetSpan(
                    child: Container(
                  color: Colors.blue,
                  width: 100.0,
                  height: 100.0,
                )),
                WidgetSpan(
                    child: Container(
                  color: Colors.yellow,
                  width: 100.0,
                  height: 100.0,
                )),
              ]),
            ),
          )
        ],
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _print,
        tooltip: 'Print',
        child: Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
```
</details>

## Result

*expected*

TextPosition：1

Offset: Offset(100.0, 0.0) 

SelectionTextBox: TextBox.fromLTRBD(100.0, 0.0, 200.0, 100.0, TextDirection.ltr)

*actual*

TextPosition：0 

Offset: Offset(0.0, 0.0) 

SelectionTextBox: TextBox.fromLTRBD(100.0, 0.0, 200.0, 100.0, TextDirection.ltr)

## flutter doctor -v

```
[√] Flutter (Channel stable, v1.12.13+hotfix.5, on Microsoft Windows [Version 10.0.18362.535], locale zh-CN)
    • Flutter version 1.12.13+hotfix.5 at D:\Flutter\flutter_source\stable 
    • Framework revision 27321ebbad (9 days ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0

 
[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at D:\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • ANDROID_HOME = D:\Android\sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b02)
    • All Android licenses accepted.

[√] Android Studio (version 3.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 29.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b02)

[√] VS Code (version 1.41.0)
    • VS Code at C:\Users\eastmoney\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.7.1

[√] Connected device (1 available)
    • MI 6 • 5a4b233b • android-arm64 • Android 8.0.0 (API 26)
```

**Comment by AlexVincent525** [comment]
cc @chunhtai

**Comment by TahaTesser** [comment]
Issue still exist

<details>
 <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel dev, v1.18.0, on Mac OS X 10.15.4 19E266, locale en-GB)
    • Flutter version 1.18.0 at /Users/taha/Code/flutter_dev
    • Framework revision 8f7327f83a (2 days ago), 2020-04-06 22:11:01 -0400
    • Engine revision 49891e0653
    • Dart version 2.8.0 (build 2.8.0-dev.20.0 1210d27678)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/taha/Code/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/taha/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4, Build version 11E146
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.0.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.43.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.9.1

 
[✓] Connected device (5 available)            
    • SM M305F   • 32003c30dc19668f                     • android-arm64  •
      Android 10 (API 29)
    • iPhone 11  • 41EB2A85-7EB4-4FD2-8978-FBBE2A681D74 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-4 (simulator)
    • macOS      • macOS                                • darwin-x64     • Mac
      OS X 10.15.4 19E266
    • Chrome     • chrome                               • web-javascript •
      Google Chrome 81.0.4044.92
    • Web Server • web-server                           • web-javascript •
      Flutter Tools

• No issues found!
```

</details>

**Comment by Jaysonss** [comment]
Any update on this issue?

**Comment by chunhtai** [comment]
Please use thumb up on the original post to bump up the priority

**Comment by chunhtai** [comment]
I know fixing this is one step toward supporting widget span in selectable text. I would like to know if there is any other use case besides the selectable text is blocked on this issue? I asking this because I plan to fix this issue and widget span support in selectable text together, but that might take a while since i still have gotten enough feedback to proceed. (See https://github.com/flutter/flutter/issues/38474#issuecomment-617916647)

**Comment by pinyin** [comment]
Still reproducible on Flutter 1.24.0-8.0.pre.229 on mobile.

**Comment by pinyin** [comment]
code:

```
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

final text = '01';

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    final richTextKey = GlobalKey();
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: RichText(
            text: TextSpan(
              children: [
                WidgetSpan(child: Container()),
                TextSpan(text: text),
              ],
            ),
            key: richTextKey),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {
          final RenderParagraph paragraph =
              richTextKey.currentContext.findRenderObject();
          print(
              paragraph.getOffsetForCaret(TextPosition(offset: 0), Rect.zero));
          print(
              paragraph.getOffsetForCaret(TextPosition(offset: 1), Rect.zero));
          print(
              paragraph.getOffsetForCaret(TextPosition(offset: 2), Rect.zero));
          print(
              paragraph.getOffsetForCaret(TextPosition(offset: 3), Rect.zero));
        },
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ),
    );
  }
}

```
On Flutter 1.24.0-8.0.pre.229, clicking the button would print:

```
I/flutter (11614): Offset(0.0, 3.0)
I/flutter (11614): Offset(0.0, 2.6)
I/flutter (11614): Offset(7.9, 2.6)
I/flutter (11614): Offset(0.0, 0.0)
```
If comment out the line with the WidgetSpan, the result would be as expected: 
```
I/flutter (11614): Offset(0.0, -0.4)
I/flutter (11614): Offset(7.9, -0.4)
I/flutter (11614): Offset(15.7, -0.4)
I/flutter (11614): Offset(0.0, 0.0)
```

**Comment by TahaTesser** [comment]
<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

final text = '01';

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    final richTextKey = GlobalKey();
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: RichText(
            text: TextSpan(
              children: [
                WidgetSpan(child: Container()),
                TextSpan(text: text),
              ],
            ),
            key: richTextKey),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {
          final RenderParagraph paragraph =
              richTextKey.currentContext.findRenderObject();
          print(
              paragraph.getOffsetForCaret(TextPosition(offset: 0), Rect.zero));
          print(
              paragraph.getOffsetForCaret(TextPosition(offset: 1), Rect.zero));
          print(
              paragraph.getOffsetForCaret(TextPosition(offset: 2), Rect.zero));
          print(
              paragraph.getOffsetForCaret(TextPosition(offset: 3), Rect.zero));
        },
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ),
    );
  }
}
```
</details>

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel stable, 1.22.5, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.22.5 at /Users/tahatesser/Code/flutter_stable
    • Framework revision 7891006299 (6 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (2 available)
    • RMX2001 (mobile)     • EUYTFEUSQSRGDA6D          • android-arm64 • Android 10 (API 29)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios           • iOS 14.3

! Doctor found issues in 1 category.
```
```bash
[✓] Flutter (Channel master, 1.26.0-2.0.pre.419, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.26.0-2.0.pre.419 at /Users/tahatesser/Code/flutter_master
    • Framework revision 913d5933c1 (4 hours ago), 2021-01-19 22:34:03 -0800
    • Engine revision b58dbc88fc
    • Dart version 2.12.0 (build 2.12.0-240.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (4 available)
    • RMX2001 (mobile)     • EUYTFEUSQSRGDA6D          • android-arm64  • Android 10 (API 29)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios            • iOS 14.3
    • macOS (desktop)      • macos                     • darwin-x64     • macOS 11.1 20C69 darwin-x64
    • Chrome (web)         • chrome                    • web-javascript • Google Chrome 87.0.4280.141

• No issues found!
```

</details>

**Comment by zmtzawqlp** [comment]
it seems that it's fixed in stable 3.3.0

**Comment by maheshmnj** [comment]
This is no longer reproducible on latest master 3.7 and stable 3.3

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

const text = '01';

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    final richTextKey = GlobalKey();
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: RichText(
            text: TextSpan(
              children: [
                WidgetSpan(child: Container()),
                const TextSpan(text: text),
              ],
            ),
            key: richTextKey),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {
          final RenderParagraph paragraph =
              richTextKey.currentContext!.findRenderObject() as RenderParagraph;
          print(paragraph.getOffsetForCaret(
              const TextPosition(offset: 0), Rect.zero));
          print(paragraph.getOffsetForCaret(
              const TextPosition(offset: 1), Rect.zero));
          print(paragraph.getOffsetForCaret(
              const TextPosition(offset: 2), Rect.zero));
          print(paragraph.getOffsetForCaret(
              const TextPosition(offset: 3), Rect.zero));
        },
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}

```
</details> 

### logs
```console
Performing hot restart...                                           89ms
Restarted application in 89ms.
Offset(0.0, 0.0)
Offset(0.0, 15.6)
Offset(7.9, 15.6)
Offset(15.7, 15.6)
```
**without WidgetSpan**

```console
Performing hot restart...                                        1,361ms
Restarted application in 1,361ms.
Offset(0.0, -0.4)
Offset(7.9, -0.4)
Offset(15.7, -0.4)
Offset(0.0, 0.0)
```

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel master, 3.7.0-3.0.pre.33, on macOS 12.6 21G115 darwin-arm64, locale en-IN)
    • Flutter version 3.7.0-3.0.pre.33 on channel master 
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5201856805 (38 minutes ago), 2022-12-05 18:27:21 -0800
    • Engine revision a309d239c4
    • Dart version 2.19.0 (build 2.19.0-463.0.dev)
    • DevTools version 2.20.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly
      to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.70.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.55.20221129

[✓] Connected device (3 available)
    • iPhone 12 Pro (mobile) • 026D5789-9E78-4AD5-B1B2-3F8D4E7F65E4 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.6 21G115 darwin-arm64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 108.0.5359.94

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
```
[✓] Flutter (Channel stable, 3.3.9, on macOS 12.6 21G115 darwin-arm, locale en-IN)
    • Flutter version 3.3.9 on channel stable at /Users/mahesh/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b8f7f1f986 (24 hours ago), 2022-11-23 06:43:51 +0900
    • Engine revision 8f2221fbef
    • Dart version 2.18.5
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.70.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.53.20221101

[✓] Connected device (3 available)
    • iPhone 12 Pro (mobile) • 026D5789-9E78-4AD5-B1B2-3F8D4E7F65E4 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.6 21G115 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 107.0.5304.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>
